test/ui/ResultLayoutTest.ts

```diff
@@ -19,6 +20,42 @@ export function ResultLayoutTest() {
       test = null;
     });
 
+    describe('with duplicate result layouts', () => {
+      let noOfActiveLayouts: number;
+      
+      beforeEach(() => {
+        noOfActiveLayouts = 0;
+      });
+
+      const addResultLayouts = (layouts: string[]) => {
```

Comment:
I think this method does a bit too much. Its named `addResultLayout` but it also sets a variable for the amount of result layouts and triggers the afterComponentsInitialization event. It makes the test appear short but we're not sure how it works because two actions are hidden in the `addResultLayout` method.

When writing tests I like to follow this structure:
given
when
then

In your case, it would be:
given: two results layout with the same name
when: populate is triggered
then: expect to have only result layout

More info [here](https://martinfowler.com/bliki/GivenWhenThen.html).